### ISSUE TITLE
Preview fails after upgrade to 3.4.1

### ISSUE THREAD

**Issue by joeknapp** [asciidoctor/asciidoctor-vscode#919]
A local document that renders a preview without issue in 3.3.1 no longer provides a preview after upgrading to 3.4.1 and re-launching VS-Code. Looking at the developer log we find:

```
[Extension Host] Unable to get Antora context for file:///c%3A/Users/redacted/Documents/redacted/redacted/redacted-v2.adoc Error: code 2: ./My Music: Access is denied. (os error 5)
./My Pictures: Access is denied. (os error 5)
./My Videos: Access is denied. (os error 5)

	at ChildProcess.<anonymous> (c:\Users\redacted\.vscode\extensions\asciidoctor.asciidoctor-vscode-3.4.1\dist\src\util\findFiles.js:75:24)
	at ChildProcess.emit (node:events:519:28)
	at maybeClose (node:internal/child_process:1105:16)
	at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

Rolling back to 3.3.1 everything works again.
	
Note that I have opened 'Documents' as my working folder and that the offending access denied errors are for links to folders outside of the 'Documents' folder and its children.

Version: 1.94.2 (user setup)
Commit: 384ff7382de624fb94dbaf6da11977bba1ecd427
Date: 2024-10-09T16:08:44.566Z
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19044

**Comment by ggrossetie** [comment]
Hey!
Where are `My Music`, `My Pictures` and `My Videos` located?

```
file:///c%3A/Users/redacted/Documents/redacted/redacted/redacted-v2.adoc
````

Are they outside of `file:///c%3A/Users/redacted/Documents`?
We are using the following glob pattern `modules/*/{attachments,examples,images,pages,partials,assets}/**`. I'm not really sure why `ripgrep` is trying to access `My Music`, `My Pictures` or `My Videos`  directories. Are you using symlinks?

/cc @alaindresse

**Comment by alaindresse** [comment]
Hi,

I also don’t understand why ripgrep tries to access folders outside of your working folder.

Do you also have that issue if you open a subfolder of 'Documents' as your working folder?

**Comment by dimsonson** [comment]
> A local document that renders a preview without issue in 3.3.1 no longer provides a preview after upgrading to 3.4.1 and re-launching VS-Code.

have same issue

**Comment by ggrossetie** [comment]
@dimsonson could you please share the developer log and also copy/paste what you get from `Help > About`?

**Comment by ryan-carpenter** [comment]
In my case, also mentioned in https://github.com/asciidoctor/asciidoctor-vscode/issues/915#issuecomment-2418402685, I found that a relative path is pointing to the wrong location.

Current file: `wrkspc/A/path/to/file.adoc`
Other file: `wrkspc/B/path/to/file.bib`
No such file: `./path/to/file.bib` (that is, `wrkspc/A/path/to/path/to/file.bib`)

Here is the error (with a redacted path).

```
  ERR code 2: ./path/to/file.bib: No such file or directory (os error 2)
: Error: code 2: ./path/to/file.bib: No such file or directory (os error 2)

at ChildProcess.<anonymous> (/home/user/.vscode-oss/extensions/asciidoctor.asciidoctor-vscode-3.4.1-universal/dist/src/util/findFiles.js:75:24)
at ChildProcess.emit (node:events:519:28)
at maybeClose (node:internal/child_process:1105:16)
at Socket.<anonymous> (node:internal/child_process:457:11)
at Socket.emit (node:events:519:28)
at Pipe.<anonymous> (node:net:339:12)
```

I also get the following error for several files, none of which refer to file.bib.

```
[Extension Host] Unable to build the Table Of Content for: wrkspc/anywhere/path/to/file.adoc Error: code 2: ./path/to/file.bib: No such file or directory (os error 2)

at ChildProcess.<anonymous> (/home/user/.vscode-oss/extensions/asciidoctor.asciidoctor-vscode-3.4.1-universal/dist/src/util/findFiles.js:75:24)
at ChildProcess.emit (node:events:519:28)
at maybeClose (node:internal/child_process:1105:16)
at Socket.<anonymous> (node:internal/child_process:457:11)
at Socket.emit (node:events:519:28)
at Pipe.<anonymous> (node:net:339:12)
```

I searched the project directory for all references to file.bib -- it exists only in a couple of attribute definitions -- and commented them out, then reloaded the window. No change. However, if I open an entirely different directory in a new (VSCodium) window, and create an AsciiDoc file there, the preview works.

**Comment by ggrossetie** [comment]
@alaindresse I'm leaning toward reverting the ripgrep feature to give us time to carefully consider each case.

**Comment by ryan-carpenter** [comment]
My project also included a symlink to file.bib, so I removed that and tried again. This time the missing file error was gone, but the preview still did not work. I have the following error for AsciiDoc and a similar one for Vale.

```
  ERR Cannot read properties of undefined (reading 'source'): TypeError: Cannot read properties of undefined (reading 'source')
	at AsciidocPreview.revive (/home/user/.vscode-oss/extensions/asciidoctor.asciidoctor-vscode-3.4.1-universal/dist/src/features/preview.js:42:49)
	at AsciidocPreviewManager.deserializeWebviewPanel (/home/user/.vscode-oss/extensions/asciidoctor.asciidoctor-vscode-3.4.1-universal/dist/src/features/previewManager.js:90:57)
	at d6.$deserializeWebviewPanel (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:156:26772)
	at Kw._doInvokeHandler (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:122963)
	at Kw._invokeHandler (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:122707)
	at Kw._receiveRequest (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:121473)
	at Kw._receiveOneMessage (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:120299)
	at gh.value (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:118398)
	at A._deliver (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:895)
	at A.fire (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:1215)
	at no.fire (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:10878)
	at gh.value (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:174:14459)
	at A._deliver (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:895)
	at A.fire (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:1215)
	at no.fire (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:10878)
	at MessagePortMain.<anonymous> (file:///usr/share/codium/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:174:12667)
	at MessagePortMain.emit (node:events:519:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:2619)
```

**Comment by ryan-carpenter** [comment]
> could you please share the developer log and also copy/paste what you get from `Help > About`?

Here is mine from VSCodium:

```
Version: 1.94.2
Release: 24286
Commit: 62f778783c52510c94e687de293bc2ad230f9a67
Date: 2024-10-12T20:14:54.465Z
Electron: 30.5.1
ElectronBuildId: undefined
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Linux x64 6.11.2-1-default
```

**Comment by alaindresse** [comment]
Makes sense. Would it be possible to put it behind a setting variable so
that those for whom it works (eg us) could keep using it ?

**Comment by ryan-carpenter** [comment]
I resolved the problem on my system by installing ripgrep. Preview now works with Asciidoctor 3.4.1.

**Comment by ggrossetie** [comment]
@ryan-carpenter interesting, it might mean that `ripgrep` is not installed/included in VSCodium. Could you please show the installation directory of VSCodium? We are expecting to find the following directory structure: `node_modules/@vscode/ripgrep/bin/rg`

**Comment by ryan-carpenter** [comment]
> @ryan-carpenter interesting, it might mean that `ripgrep` is not installed/included in VSCodium. Could you please show the installation directory of VSCodium? We are expecting to find the following directory structure: `node_modules/@vscode/ripgrep/bin/rg`

Actually, ripgrep is included in `/usr/share/codium/resources/app/node_modules/@vscode`, though I did not know this before installing it directly from the openSUSE Tumbleweed repo.

The openSUSE version is ripgrep 14.1.1. and `node_modules/@vscode/ripgrep/bin/rg --version` returns 
```
ripgrep 13.0.0 (rev af6b6c543b)
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

The ripgrep directory and most of its subdirectories under `/usr/share/codium ...` were last modified 16 Oct 2024, which coincides with the date that I upgraded the Asciidoctor extension, so it looks like the dependency was installed but there appears to be an issue with ripgrep version 13.0.0.
